gridsome/lib/graphql/schema/types/image.js

```diff
@@ -0,0 +1,31 @@
+const fs = require('fs')
+const path = require('path')
+
+const { GraphQLJSON } = require('../../graphql')
+const { SUPPORTED_IMAGE_TYPES } = require('../../../utils/constants')
+
+exports.isImage = value => {
+  const extName = path.extname(value)
+
+  if (!(typeof value === 'string' || value instanceof String)) {
+    return false
+  }
+
+  if (!SUPPORTED_IMAGE_TYPES.includes(extName.toLowerCase())) {
+    return false
+  }
+
+  if (path.basename(value) === value) {
+    return false
+  }
+
+  return true
+}
+
+exports.imageType = {
+  type: GraphQLJSON,
+  args: {},
+  resolve: async (fields, { }, context, { fieldName }) => {
+    return await context.queue.add(fields[fieldName]);;
```

Comment:
The `g-image` component only need `src`, `srcset`, `sizes`, `size` and `dataUri` from the queue results. So you can probably return those properties.